### PR TITLE
[EOS-25904] support 'data_node' or 'storage_node' as node type

### PR DESCRIPTION
**Pull request by JugalPatil**
# Problem Statement
- add 'data_node' and 'storage_node' as valid node types for F99C integration.

# Design
-  Changes for separate s3 pod

# Coding
-  Coding conventions are followed and code is consistent

# Testing 
-  Test Cases cover Happy Path, Non-Happy Path and Scalability
-  Testing was performed with RPM

# Review Checklist 
- https://jts.seagate.com/browse/EOS-25904

Signed-off-by: Jugal Patil <jugal.patil@seagate.com>

### CODE REVIEW

**Review by nkommuri — APPROVED** [review]

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 1]
```diff
@@ -414,13 +414,13 @@ def motr_config_k8(self):
         raise MotrError(errno.EINVAL, "libfabric is not up.")
 
     # Update motr-hare keys only for storage node
```
Comment (mehjoshi) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:416]: update comment to say storage or data nodes.

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py [hunk 2]
```diff
@@ -414,13 +414,13 @@ def motr_config_k8(self):
...
-    if self.node['type'] == 'storage_node':
+    if (self.node['type'] == 'storage_node' or self.node['type'] == 'data_node'):
         update_motr_hare_keys(self, self.nodes)
 
     execute_command(self, MOTR_CONFIG_SCRIPT, verbose = True)
 
     # Update be_seg size only for storage node
```
Comment (mehjoshi) [inline scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py:422]: update comment to say storage or data nodes.

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py (hunk 1 above)
Comment (JugalPatil) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py]: comment updated

File: scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py (hunk 2 above)
Comment (JugalPatil) [reply scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py]: comment on line 422 updated

**Review by mehjoshi — APPROVED** [review]

**Review by madhavemuri — APPROVED** [review]
https://github.com/Seagate/cortx-hare/pull/1908/files
Need to use logic similar to above  PR instead of data and storage node support.

This can be a separate change.

**Pull request merged**